### PR TITLE
feat/create password reset

### PR DESCRIPTION
**Pull request by Myles181**
## Title
Implement Password Reset Functionality via Email OTP

## Description
Implemented a password reset feature using FastAPI. This feature allows users to request a password reset by providing their email address. An OTP code is generated and sent to the user's email address, which they can use to reset their password. The endpoint does not change the password directly; it only handles OTP generation and email sending.

## Related Issue
This pull request is related to the following issue: [#IssueNumber](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/297)

## Motivation and Context
This change is required to provide a secure and user-friendly way for users to reset their passwords in case they forget it. It enhances the user experience by allowing them to receive an OTP code via email, ensuring that only the rightful owner of the email address can reset the password.

## How Has This Been Tested?
- The endpoint was tested using Postman to ensure that it correctly sends an OTP to the provided email address.
- Various scenarios were tested, including:
  - Valid email address
  - Non-existent email address
  - Missing email field
- The OTP generation and email sending functionalities were tested to ensure they work as expected.
- The database was checked to ensure OTP records are created with correct expiration times.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Additional Notes
- The `OTP` model was created to store OTP codes with fields for email, OTP code, creation time, and expiration time.
- The `password_reset` endpoint was added to handle the password reset requests.
- Custom exception handling was implemented to manage various error scenarios.

### CODE REVIEW

File: api/v1/schemas/password_reset.py
Comment (joboy-dev) [inline api/v1/schemas/password_reset.py:1]: Change from EMAILSCHEMA to EnailSchema

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/v1/schemas/password_reset.py
Comment (Myles181) [reply api/v1/schemas/password_reset.py:1]: Changes are Done 👍🏾✅

File: api/v1/routes/password_reset.py
Comment (joboy-dev) [inline api/v1/routes/password_reset.py:1]: There are already custom exception handlers defined
Do not make db calls in routes. That is the job of services

File: api/v1/models/base.py
Comment (joboy-dev) [inline api/v1/models/base.py:1]: I do not know the purpose of this. There has to be a cleaner, less confusing way of handling this

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Comment by Myles181** [comment]
Please I've removed that piece of code 
It was to create a relationship with the user and password_reset in another table.

**Pull request closed without merging**